pkg/controllers/machinesync/machine_sync_controller_test.go

```diff
@@ -91,35 +140,234 @@ var _ = Describe("MachineSync Reconciler", func() {
+								HaveField("Message", Equal("Successfully synchronized CAPI Machine to MAPI")),
+							))),
+					)
+				})
+			})
+
+			Context("when the CAPI machine does exist", func() {
+				BeforeEach(func() {
+					capiMachine = capiMachineBuilder.Build()
+					Expect(k8sClient.Create(ctx, capiMachine)).Should(Succeed())
+				})
+
+				It("should update the synchronized condition on the MAPI machine to True", func() {
+					Eventually(k.Object(mapiMachine), timeout).Should(
+						HaveField("Status.Conditions", ContainElement(
+							SatisfyAll(
+								HaveField("Type", Equal(consts.SynchronizedCondition)),
+								HaveField("Status", Equal(corev1.ConditionTrue)),
+								HaveField("Reason", Equal("ResourceSynchronized")),
+								HaveField("Message", Equal("Successfully synchronized CAPI Machine to MAPI")),
+							))),
+					)
+				})
+			})
+		})
+
+		Context("when the MAPI machine has MachineAuthority set to Migrating", func() {
+			BeforeEach(func() {
+				By("Creating the CAPI and MAPI machines")
+				// We want a difference, so if we try to reconcile either way we
+				// will get a new resourceversion
+				mapiMachine = mapiMachineBuilder.Build()
+				capiMachine = capiMachineBuilder.Build()
+
+				Expect(k8sClient.Create(ctx, capiMachine)).Should(Succeed())
+				Expect(k8sClient.Create(ctx, mapiMachine)).Should(Succeed())
+
+				By("Setting the AuthoritativeAPI to Migrating")
+				Eventually(k.UpdateStatus(mapiMachine, func() {
+					mapiMachine.Status.AuthoritativeAPI = machinev1beta1.MachineAuthorityMigrating
+				})).Should(Succeed())
+			})
+
+			It("should not make any changes to either machine", func() {
+				mapiResourceVersion := mapiMachine.GetResourceVersion()
+				capiResourceVersion := capiMachine.GetResourceVersion()
+				Consistently(k.Object(mapiMachine), timeout).Should(
+					HaveField("ResourceVersion", Equal(mapiResourceVersion)),
+				)
+				Consistently(k.Object(capiMachine), timeout).Should(
+					HaveField("ResourceVersion", Equal(capiResourceVersion)),
+				)
+			})
+		})
+
+		Context("when the MAPI machine has MachineAuthority not set", func() {
+			BeforeEach(func() {
+				By("Creating the CAPI and MAPI Machines")
+				mapiMachine = mapiMachineBuilder.Build()
+				capiMachine = capiMachineBuilder.Build()
+
+				Expect(k8sClient.Create(ctx, capiMachine)).Should(Succeed())
+				Expect(k8sClient.Create(ctx, mapiMachine)).Should(Succeed())
+
+				By("Setting the AuthoritativeAPI to Migrating")
+				Eventually(k.UpdateStatus(mapiMachine, func() {
+					mapiMachine.Status.AuthoritativeAPI = ""
+				})).Should(Succeed())
+			})
+
+			It("should not make any changes", func() {
+				resourceVersion := mapiMachine.GetResourceVersion()
+				Consistently(k.Object(mapiMachine), timeout).Should(
+					HaveField("ResourceVersion", Equal(resourceVersion)),
+				)
+			})
+		})
+
+		Context("when the MAPI machine set does not exist and the CAPI machine set does", func() {
```

Comment:
machine set in this description 
```suggestion
		Context("when the MAPI machine does not exist and the CAPI machine does", func() {
```